test(openchallenges): improve organization service model entity unit test coverage

mdsage1 opened the pull request:
## Description

Sonar unit organization service test coverage has been increased to 45% for model/entity package by writing unit tests for a single class, OrganizationEntity.


## Related Issue

#2063 
#1528 
#2348 
#2388 

## Changelog

See `files Changed` section

These changes have increased the Sonar coverage from the 15% achieved with the exception unit test PR #2348 (9.45% overall). These tests were written last week and debugged this week. The PR has been released after the approval of the exception unit test PR on 2023-12-04.

## Preview

Overall sonar coverage has increased to 50%, from 0%, for organization service model entity.

The Sonar coverage increase for model/mapper:
![image](https://github.com/Sage-Bionetworks/sage-monorepo/assets/122999770/dff666aa-564a-4458-a77e-e9c43dcd9d4e)

The goal was to get organization-service > 30% and I will confirm this once the PR has been merged. Given that the Sonar coverage calculation is 15% higher than the intended coverage it's possible that the [Code Quality Dashboard](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/CODE_QUALITY.md) will reflect 30% code coverage with the updates made to this class.

mdsage1 commented:
@tschaffter Do you feel this covers #1528? Or were you thinking about another class when creating this?

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,227 @@
+            .updatedAt(updatedAt)
+            .acronym(acronym)
+            .build();
+    Long retrievedId = organizationEntity.getId();
```
Two name terms used often together are "actual" and "expected" value. Here you could replace "retrieved" by "actual". You could optionally define above "expectedName", etc. but it's fine with me to keep using "name", "email", etc.

As usual, please consider other places where this comment may apply (I may occasionally no longer mention this).

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,227 @@
+
+    OrganizationEntity organizationEntity = new OrganizationEntity();
+
+    organizationEntity.setId(1L);
+    organizationEntity.setName("Test Organization");
+    organizationEntity.setEmail("test@example.com");
+    organizationEntity.setLogin("testorg");
+    organizationEntity.setDescription("Test description");
+    organizationEntity.setAvatarKey("avatarKey");
+    organizationEntity.setWebsiteUrl("https://example.com");
+    organizationEntity.setChallengeCount(5);
+    organizationEntity.setCreatedAt(createdAt);
+    organizationEntity.setUpdatedAt(updatedAt);
+    organizationEntity.setAcronym("TO");
+
+    OrganizationEntity organizationEntity2 = new OrganizationEntity();
+    organizationEntity2.setId(1L);
+    organizationEntity2.setName("Test Organization");
+    organizationEntity2.setEmail("test@example.com");
+    organizationEntity2.setLogin("testorg");
+    organizationEntity2.setDescription("Test description");
+    organizationEntity2.setAvatarKey("avatarKey");
+    organizationEntity2.setWebsiteUrl("https://example.com");
+    organizationEntity2.setChallengeCount(5);
+    organizationEntity2.setCreatedAt(createdAt);
+    organizationEntity2.setUpdatedAt(updatedAt);
+    organizationEntity2.setAcronym("TO");
```
This is a good opportunity to start using a "@BeforeAll" (see this [article](https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall)) to define objects ONCE and then use them in more than one test.

mdsage1 replied:
Added `@BeforeAll` to the code to initialized the properties of each organization entity that is generated.

mdsage1 commented:
Changes implemented:
1. Updated organizationEntity to entity in tests with a single entity 
2. Changed all occurrences of `retrieved` in variable names to `actual`
3. TBD on how to implement @BeforeAll w/o throwing errors

tschaffter commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,163 @@
+
+  @Test
+  public void
+      OrganizationEntityBuilderArguments_ShouldReturnArgumentsPassed_WhenOrganizationEntityBuilt() {
```
Review title: the "BuilderArguments" don't "return" anything. You could say that the org entity returns the arguments passed when built with the class builder.

mdsage1 replied:
Changed the name to `OrganizationEntity_ShouldReturnArgumentsPassed_WhenOrganizationEntityBuiltWithClassBuilder`

mdsage1 commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,163 @@
+
+public class OrganizationEntityTest {
+
+  private static Long id;
```
addressed in the `Define these values at the same time you define the class member variables for a more concise code.` change request.

mdsage1 commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,163 @@
+
+  @BeforeEach
+  public void setup() {
+    id = 1L;
+    name = "Test Organization";
+    email = "test@example.com";
+    login = "testorg";
+    avatarKey = "avatarKey";
+    websiteUrl = "https://example.com";
+    challengeCount = 5;
+    description = "Test description";
+    createdAt = OffsetDateTime.now();
+    updatedAt = OffsetDateTime.now();
+    acronym = "TO";
```
defined the values at the same time as defining the class variables and removed `static`.

mdsage1 commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,163 @@
+  }
+
+  @Test
+  public void HashCode_ShouldBuildHashCodeOfArgumentsForOrganizationEntity_WhenArgumentsPassed() {
```
Changed to `HashCode_ShouldBuildHashCodeOfOrganizationEntity_WhenArgumentsPassed`

mdsage1 commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,163 @@
+
+  @Test
+  public void
+      OrganizationEntityMethodsToSetArguments_ShouldReturnArguments_WhenOrganizationEntityArgumentsPassed() {
```
Changed the name to `OrganizationEntityGetters_ShouldReturnExpectedValues_WhenArgumentsPassed`

mdsage1 commented:
Organization entity properties, login and email, were removed during PR #2399 which caused the OrganizationEntityTest.java unit tests to fail. Removed the unit tests. Sonar coverage was reduced to 48%.
![image](https://github.com/Sage-Bionetworks/sage-monorepo/assets/122999770/85ac03d6-72d2-4bba-8e36-4560f3f8ae8e)

mdsage1 commented on apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:
```diff
@@ -0,0 +1,137 @@
+  }
+
+  @Test
+  public void EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects() {
```
Implemented the following changes:

1. Changed the unit test name from EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects to EqualsObject_ShouldBeTheSame_WhenPassedTwoIdenticalObjects
2. Separated that test into 2 smaller tests 1 test differentiates between two different objects and the other recognizes two objects as being identical.

mdsage1 commented:
@tschaffter This is ready for you to review. I made the changes you asked for:

> Implemented the following changes:
> 
> 1. Changed the unit test name from EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects to EqualsObject_ShouldBeTheSame_WhenPassedTwoIdenticalObjects
> 2. Separated that test into 2 smaller tests 1 test differentiates between two different objects and the other recognizes two objects as being identical.
